packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -705,7 +705,7 @@ class _PageViewState extends State<PageView> {
           }
         } else if (notification.depth == 0 && widget.pageDidChange != null && notification is ScrollEndNotification) {
           final PageMetrics metrics = notification.metrics;
-          final int currentPage = metrics.page.round();
+          final int currentPage = metrics.page.toInt();
```

Comment:
This seems odd now. Due to how doubles work, `metrics.page` may be stuck at say 2.999999 even though the transition has completed. In that case we would no longer get a notification that we've reached page three?

Reply:
You are right.I will revert this line and disable this feature when snapping is off, so `pageDidChange` would never trigger when `pageSnapping`'s value is false.It's that OK?